utils/scripts/ocm/ocm.py

```diff
@@ -844,6 +848,73 @@ def update_notification_email_address(self, addon_name, email_address, exit_on_f
             else:
                 return False
 
+    def get_latest_osd_candidate_version(self, osd_major_version, osd_minor_version):
+        """
+        get the latest  candidate version
+        Args:
+            osd_major_version (str):  Major version of the release. For example in 4.8.10 : 4 is major version
+            osd_minor_version (str):  Major version of the release. For example in 4.8.10 : 4 is minor version
+            Example 4.8 = osd_major_version.osd_minor_version       
+        """
+        cmd = ("ocm list versions --channel-group  candidate |"
+               " grep  ^{}.{}|tail -1".format(osd_major_version, osd_minor_version))
+        ret = execute_command(cmd)
+        if ret is None:
+            log.info("Failed  to get latest version of ODS ")
+        return ret
+
+    def get_all_osd_versions(self):
+        """
+        gets the latest osd version and convert it in list
+        Example:
+            {'4': {'4.8': '4.8.39', '4.9': '4.9.33', '4.10': '4.10.14'}, '5': {'': ''}}
+        """
+        # Dict that will be converted into json file
+        latest_osd_versions_data = {}
+        dic = {}
+        for candidate_version in range(int(self.osd_minor_version_start), int(self.osd_minor_version_end)):
+            version = (self.get_latest_osd_candidate_version(
+                self.osd_major_version, candidate_version)).split("-")[0]
+            dic[".".join(version.split(".")[:2])] = version
+            latest_osd_versions_data[str(self.osd_major_version)] = dic
+        log.info(latest_osd_versions_data)
+        return latest_osd_versions_data
+
+    def compare_with_old_version_file(self):
+        """
+        Compares the latest osd version in the json  file and Updates
+        Run key
```

Comment:
could you explain this a bit more? otherwise without reading the cose it's difficult to understand what Run key means

Reply:
ack Done